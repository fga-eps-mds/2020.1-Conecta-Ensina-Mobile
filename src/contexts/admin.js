import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';

export const AdmContext = createContext({});

function AdmProvider({children}) {
  const [students, setStudents] = useState(null);
  const [userDB, setUserDB] = useState(null);
  const [teacher, setTeacher] = useState(null);

  const {Host, user} = useContext(AuthContext);

  async function getProfessorList() {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);

    console.log(Host);

    try {
      const data = await fetchResponse.json();
      setStudents(data.data.student);
    } catch (error) {
      return error;
    }
  }

  async function statusUpdate(id) {
    const settings = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        status: 1,
        agentRole: user.role,
      }),
    };
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    try {
      const data = await fetchResponse1.json();
      console.log('Success:', data);
    } catch (error) {
      console.error('Error:', error);
    }
  }

  async function getProfessoUser(id) {
    const fetchResponse = await fetch(`${Host}/api/user/${id}`);
    try {
      const data = await fetchResponse.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: data.data.user.password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      setUserDB(usuario);
      try {
        let response = await fetch(`${Host}/api/teacher/${id}`);
        const data = await response.json();
        let professor = {
          photo: data.data.teacher.photo,
          video: data.data.teacher.video,
          graduation_area: data.data.teacher.graduation_area,
          degree: data.data.teacher.degree,
          bank: data.data.teacher.bank,
          agency: data.data.teacher.agency,
          account: data.data.teacher.account,
        };
        setTeacher(professor);
      } catch (error) {
        console.error(error);
      }
    } catch (error) {
      return error;
    }
  }

  return (
    <AdmContext.Provider
      value={{
        students,
        userDB,
        teacher,
        getProfessorList,
        statusUpdate,
        getProfessoUser,
      }}>
      {children}
    </AdmContext.Provider>
  );
}

export default AdmProvider;
