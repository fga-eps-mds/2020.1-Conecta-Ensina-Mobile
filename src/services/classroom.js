export async function getNextClassroom(Host) {
  const fetchResponse = await fetch(
    `${Host}/api/classroom/nextClass/3bd7c190-ce64-4827-8c0c-58cfef45ad9f`,
  );
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function getUserClassroom(Host, id) {
  const fetchResponse = await fetch(`${Host}/api/classroom/userClasses/${id}`);
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function createClass(user, teacher, filter, student, Host) {
  const response = await fetch(`${Host}/api/classroom/create`, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      agentRole: user.role,
      teacher: teacher,
      student: user.id,
      grade: student.grade,
      subject: filter.subject,
      dtclass: filter.dtClass,
      duration: filter.duration,
      cep: student.cep,
      number: student.number,
      details: 'oi',
    }),
  });
  try {
    const data = await response.json();
    console.log('Success: ', data);
    return data;
  } catch (error) {
    console.log(error);
  }

  console.log(response);
}
