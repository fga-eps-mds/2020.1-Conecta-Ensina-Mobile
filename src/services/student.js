export async function getStudent(Host, id) {
  const response = await fetch(`${Host}/api/student/${id}`);
  try {
    const data = await response.json();

    //setSubjects(data.data.subject);
    return data.data;
  } catch (error) {
    return error;
  }
}