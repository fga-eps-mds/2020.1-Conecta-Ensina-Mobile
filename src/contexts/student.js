import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Student from '../services/student';

export const StudentContext = createContext({});

export default function UserProvider({children}) {
  const [student, setStudent] = useState({});
  const {Host} = useContext(AuthContext);

  async function getStudent(id) {
    const response = await Student.getStudent(Host, id);
    if (student !== response) {
      console.log(response);
      setStudent(response);
    }
  }

  return (
    <StudentContext.Provider value={{student, getStudent}}>
      {children}
    </StudentContext.Provider>
  );
}
