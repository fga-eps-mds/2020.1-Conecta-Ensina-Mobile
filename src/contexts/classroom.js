import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Class from '../services/classroom';
import * as Student from '../services/student';
import * as Address from '../services/findAddress';
import * as GeoCode from '../services/geocode';
import {FiltersContext} from '../contexts/filters';
import {StudentContext} from '../contexts/student';

export const ClassroomContext = createContext({});

export default function ClassroomProvider({children}) {
  const [classroom, setClassroom] = useState({});
  const [firstClass, setFirstClass] = useState(null);
  const [classes, setClasses] = useState(null);
  const {Host, user} = useContext(AuthContext);
  const {stackStudents} = useContext(StudentContext);
  const {filter} = useContext(FiltersContext);
  const [statusClasses, setStatusClasses] = useState([]);
  const [statusClass, setStatusClass] = useState({});
  const [coordenates, setCoordenates] = useState({});

  async function loadNextClass() {
    const response = await Class.getNextClassroom(Host);
    if (classroom !== response) {
      setFirstClass(response);
    }
  }
  async function createClass(teacher) {
    //console.log('Esse aqui Ã© o professor: ' + teacher);
    const student = await Student.getStudent(Host, user.id);
    const response = await Class.createClass(
      user,
      teacher,
      filter,
      student.student,
      Host,
    );
  }

  async function loadUserClasses() {
    const response = await Class.getUserClassroom(Host, user.id);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function getClassroom(status) {
    let teacher, student;
    if (user.role === 2) {
      student = '';
      teacher = user.id;
    } else {
      student = user.id;
      teacher = '';
    }
    const response = await Class.getClassroom(student, teacher, status, Host);

    if (classroom !== response) {
      //console.log(response);
      setStatusClasses(response);
    }
  }

  async function updateStatusClassroom(id) {
    let response = await Class.readClass(Host, id);
    //console.log(response.status);
    if (response.status === 1) {
      response = await Class.updateStatusClassroom(id, 2, Host);
      //console.log(response.message);
    } else if (response.status === 2) {
      response = await Class.updateStatusClassroom(id, 3, Host);
      //console.log(response.message);
    } else if (response.status === 3) {
      response = await Class.updateStatusClassroom(id, 4, Host);
      //console.log(response.message);
    } else if (response.status === 4) {
      response = await Class.updateStatusClassroom(id, 5, Host);
      //console.log(response.message);
    }
  }
  async function loadStatusClasses(status) {
    const response = await Class.getStatusClassroom(Host, user.id, status);
    if (statusClasses !== response) {
      await setStatusClasses(response);
    }
    //console.log(response);
    return response;
  }

  async function loadStatusClassesStudents(status) {
    const response = await loadStatusClasses(status);
    let studentList = await {list: []};
    await Promise.all(
      response.map(async (item) => {
        const student = await Student.getStudent(Host, item.student);
        studentList.list = await studentList.list.concat(student);
      }),
    );
    await stackStudents(studentList.list);
  }

  async function readClass(id) {
    const response = await Class.readClass(Host, id);
    const address = await Address.findAddress(response.cep);
    let responseClass = {
      id: response.role,
      status: response.status,
      teacher: response.teacher,
      student: response.student,
      grade: response.grade,
      subject: response.subject,
      dtclass: response.dtClass,
      duration: response.duration,
      cep: response.cep,
      number: response.number,
      details: response.details,
      address: address,
    };

    setClassroom(responseClass);
  }

  async function getClass() {
    const response = await Class.getClass(Host, user);
    setClasses(response);
  }

  async function updateStatusClasses(id, status) {
    const response = await Class.updateStatus(id, status, Host);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function geoCode(cep) {
    const response = await GeoCode.geocode(cep);
    if (coordenates !== response) {
      //console.log(response.results[0].geometry.location);
      setCoordenates(response.results[0].geometry.location);
    }
  }

  return (
    <ClassroomContext.Provider
      value={{
        classroom,
        firstClass,
        classes,
        loadNextClass,
        createClass,
        loadUserClasses,
        getClassroom,
        updateStatusClassroom,
        loadStatusClasses,
        readClass,
        statusClasses,
        statusClass,
        setStatusClass,
        setStatusClasses,
        Host,
        getClass,
        updateStatusClasses,
        loadStatusClassesStudents,
        geoCode,
        coordenates,
      }}>
      {children}
    </ClassroomContext.Provider>
  );
}
