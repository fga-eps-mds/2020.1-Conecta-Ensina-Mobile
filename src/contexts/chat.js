import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';
import * as Chat from '../services/chat';
import { ClassroomContext } from './classroom';

export const chatContext = createContext({});

export default function ChatProvider({children}) {
  const {Host, user} = useContext(AuthContext);
  const {classroom} = useContext(ClassroomContext);
  const [chat, setChat] = useState(null);

  useEffect(() => {
      readChat()
  }, [])

  async function createChat(text) {
    const response = await Chat.createMessage(
        user.id, 
        text, 
        classroom.id, 
        classroom.teacher, 
        classroom.student, 
        Host)
  }

  async function readChat() {
    console.log(classroom)
    const response = await Chat.getChat(Host, classroom.id);
    if (chat !== response) {
      setChat(response);
    }
  }
  return (
    <chatContext.Provider
      value={{
        readChat,
        createChat,
        chat
      }}>
      {children}
    </chatContext.Provider>
  );
}
