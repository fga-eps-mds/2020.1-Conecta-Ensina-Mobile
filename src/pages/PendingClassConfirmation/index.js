import React, {useContext} from 'react';
import Theme from '../../../Theme';
import Background1 from '../../components/Background1';
import CustomText from '../../components/CustomText';
import {
  Container,
  ContainerGrande,
  ContainerButton,
  ButtonConfirmar,
  ButtonRecusar,
  ContainerComplain,
  ComplainButton,
  InfoContainer,
} from './styles';
import {StudentContext} from '../../contexts/student';
import {ClassroomContext} from '../../contexts/classroom';

export default function PendingClassConfirmation({route, navigation}) {
  const {item} = route.params;
  const {getStudent, student} = useContext(StudentContext);
  const {updateStatusClasses} = useContext(ClassroomContext);
  const params = student.id;

  return (
    <Theme>
      <Background1 navigation={navigation} page={'TeacherProfile2'}>
        <Container>
          <InfoContainer>
            <ContainerGrande>
              <CustomText white> Data: {item.dtclass}</CustomText>
            </ContainerGrande>
            <ContainerGrande>
              <CustomText white>Distância: {student.cep}</CustomText>
            </ContainerGrande>
            <ContainerGrande>
              <CustomText white>Duração: {item.duration}</CustomText>
            </ContainerGrande>
            <ContainerGrande>
              <CustomText white>Matéria: {item.subject}</CustomText>
            </ContainerGrande>
            <ContainerGrande>
              <CustomText white>
                Nome: {student.User.firstName + ' ' + student.User.lastName}
              </CustomText>
            </ContainerGrande>
            <ContainerGrande>
              <CustomText white>Instituição: {student.institution}</CustomText>
            </ContainerGrande>
            <ContainerGrande>
              <CustomText white>Série: {student.grade}</CustomText>
            </ContainerGrande>
            <ContainerGrande>
              <CustomText white>Especial: {student.special}</CustomText>
            </ContainerGrande>
            <ContainerGrande>
              <CustomText white>Descrição: {student.description}</CustomText>
            </ContainerGrande>
            <ContainerGrande>
              <CustomText white>Detalhe:{student.details}</CustomText>
            </ContainerGrande>
            <ContainerComplain>
              <ComplainButton
                testID="Reportar"
                onPress={async () => {
                  await getStudent(student.id);
                  navigation.navigate('FeedbackTeacher', {params});
                }}>
                <CustomText white>Reportar</CustomText>
              </ComplainButton>
            </ContainerComplain>
          </InfoContainer>
        </Container>
        <ContainerButton>
          <ButtonConfirmar
            testID="Aceitar"
            onPress={() => {
              updateStatusClasses(item.id, 1);
              navigation.navigate('PendingClass');
            }}>
            <CustomText white bigSmall>
              Aceitar
            </CustomText>
          </ButtonConfirmar>
          <ButtonRecusar
            testID="Recusar"
            onPress={async () => {
              await updateStatusClasses(item.id, -1);
              navigation.navigate('PendingClass');
            }}>
            <CustomText white bigSmall>
              Recusar
            </CustomText>
          </ButtonRecusar>
        </ContainerButton>
      </Background1>
    </Theme>
  );
}
