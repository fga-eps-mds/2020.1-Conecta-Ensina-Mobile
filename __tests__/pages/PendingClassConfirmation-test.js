import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import PendingClassConfirmation from '../../src/pages/PendingClassConfirmation';
import {StudentContext} from '../../src/contexts/student';
import {ClassroomContext} from '../../src/contexts/classroom';

describe('Testing Pending Class Page', () => {
  const navigation = {
    navigate: jest.fn(),
  };

  const student = {
    student: {
      cep: '',
      institution: '',
      grade: '',
      special: '',
      description: '',
      details: '',
    },
    user: {
      firstName: '',
      lastName: '',
    },
  };
  const getStudent = jest.fn();
  const updateStatusClasses = jest.fn();
  const route = {
    params: {
      item: {
        dtclass: '',
        cep: '',
        duration: '',
        grade: '',
        details: '',
        student: '',
        id: '',
      },
    },
  };

  test('Should renders page', () => {
    const tree = render(
      <StudentContext.Provider value={{getStudent, student}}>
        <ClassroomContext.Provider value={{updateStatusClasses}}>
          <PendingClassConfirmation navigation={navigation} route={route} />
          ).toJSON();
        </ClassroomContext.Provider>
      </StudentContext.Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
  test("Should press 'Reportar' button ", async () => {
    const {getByTestId} = render(
      <StudentContext.Provider value={{getStudent, student}}>
        <ClassroomContext.Provider value={{updateStatusClasses}}>
          <PendingClassConfirmation navigation={navigation} route={route} />
        </ClassroomContext.Provider>
      </StudentContext.Provider>,
    );

    const button = getByTestId('Reportar');

    await fireEvent.press(button);
  });
  test("Should press 'Aceitar' button ", async () => {
    const {getByTestId} = render(
      <StudentContext.Provider value={{getStudent, student}}>
        <ClassroomContext.Provider value={{updateStatusClasses}}>
          <PendingClassConfirmation navigation={navigation} route={route} />
        </ClassroomContext.Provider>
      </StudentContext.Provider>,
    );

    const button = getByTestId('Aceitar');

    await fireEvent.press(button);
  });
  test("Should press 'Recusar' button ", async () => {
    const {getByTestId} = render(
      <StudentContext.Provider value={{getStudent, student}}>
        <ClassroomContext.Provider value={{updateStatusClasses}}>
          <PendingClassConfirmation navigation={navigation} route={route} />
        </ClassroomContext.Provider>
      </StudentContext.Provider>,
    );

    const button = getByTestId('Recusar');

    await fireEvent.press(button);
  });
});
