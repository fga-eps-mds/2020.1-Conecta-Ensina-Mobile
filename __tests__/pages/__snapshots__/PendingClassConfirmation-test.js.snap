// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pending Class Page Should renders page 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#184E79",
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row-reverse",
          "flexGrow": 10,
          "flexShrink": 1,
          "marginBottom": "5%",
          "opacity": 1,
          "paddingBottom": "10%",
          "paddingLeft": "10%",
          "paddingRight": "10%",
          "paddingTop": "10%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 20,
              "textAlign": "center",
            },
          ]
        }
        white={true}
      >
        Selecione as disciplinas em que deseja lecionar
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderTopLeftRadius": 20,
            "borderTopRightRadius": 20,
            "display": "flex",
            "flexBasis": 0,
            "flexGrow": 78,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <RCTScrollView
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": "5%",
              "width": "100%",
            },
          ]
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": "90%",
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                 Data: 
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                Distância: 
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                Duração: 
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                Matéria: 
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                Nome: 
                 
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                Instituição: 
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                Série: 
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                Especial: 
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                Descrição: 
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 9,
                    "height": "9%",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "width": "78%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "textAlign": "center",
                    },
                  ]
                }
                white={true}
              >
                Detalhe:
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "marginTop": "8%",
                  },
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#E46270",
                    "borderRadius": 8,
                    "height": 31,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 138,
                  }
                }
                testID="Reportar"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "Roboto",
                        "fontSize": 20,
                        "textAlign": "center",
                      },
                    ]
                  }
                  white={true}
                >
                  Reportar
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-around",
              "width": "90%",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2D9D49",
              "borderRadius": 8,
              "height": 35,
              "justifyContent": "center",
              "opacity": 1,
              "width": 150,
            }
          }
          testID="Aceitar"
        >
          <Text
            bigSmall={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            Aceitar
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E46270",
              "borderRadius": 8,
              "height": 35,
              "justifyContent": "center",
              "opacity": 1,
              "width": 150,
            }
          }
          testID="Recusar"
        >
          <Text
            bigSmall={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            Recusar
          </Text>
        </View>
      </View>
    </View>
  </View>,
  ").toJSON();",
]
`;
